backend/DomainModels/Department.cs

```diff
@@ -0,0 +1,13 @@
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace backend.DomainModels;
+
+public class Department
+{
+    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public required int Id { get; set; }
```

Comment:
department is today a string, like "trondheim" which is used as Id.
We should do something similar here, but maybe make it clear that it is a key and not a value.